### DISCUSSION TITLE
Reached the free limit, can't access hours of work done

### DISCUSSION THREAD

**Discussion by brawaru** [community/community#58024] [Codespaces] [answered]
### Select Topic Area

Question

### Body

I had been contributing to an open source project for which I don't really have a powerful enough machine, so I thought I can use Codespaces. Unfortunately, the project requires a 4-core 8 GB machine type, so it very quickly reached the free quota.

While I kept in mind that GitHub said that I am about to reach the limit, and exported my changes to a branch; before working on that branch I had other changes that I've stashed, that contained several hours of work. GitHub only provides an option to export changes made to the active branch to a separate remote branch, without a word about stashes or anything else.

I contacted support hoping they can at least help me exporting the stash to a patch file, but they said ‘Sorry, we can't do anything here, but hey, paying is always the option’. And unfortunately, in my situation paying is not really an option due to several reasons outside of my control.

Can GitHub do anything to help me out in this case? This is not the first time I got completely locked out (the other time was after release, before there was even an option to export changes, where GitHub just deleted my Codepages, but thankfully they didn't have anything important on them), and it's definitely a great mistake of mine to forget to export any changes, knowing Codespace will go away, but still... It's not like that wouldn't happen to anyone else, we can't remember everything all the time.

If I can make a suggestion how you can make this experience better: disable Codespace on 99% (or at least 10-20 minutes left) of usage, and once user tries to activate Codespace again, scream at them that they only so much usage left, so they should export their changes immediately, unless they plan to pay. I don't think adding option to export the full state of machine would be suitable, because there are a lot of irrelevant files that can weight GBs, so at least that could help prevent such lockouts.

**Comment by cwndrws** [comment, marked as answer]
Thanks for writing in. I'm sorry that happened to you. Unfortunately due to the way our exporting feature works (creating a branch on your repo with your changes), we don't have a way of exporting anything other than the working branch, like you mentioned. I don't believe there's anything we can do to retrieve your stashed changes as this point.

This downside of our current exporting feature is known and I've added your feedback to the internal discussion of the feature, though there are no plans to make any changes to the export functionality in the near term.

To ensure the codespace containing your changes doesn't get deleted, you can follow [these instructions](https://docs.github.com/en/codespaces/customizing-your-codespace/configuring-automatic-deletion-of-your-codespaces#avoiding-automatic-deletion-of-codespaces) to "keep" your codespace, preventing it from being auto-deleted.

**Reply by brawaru** [reply]
Well, I don't think I'm going to wait another 17 days to continue my work. I guess I'll have to bear with it and start from scratch. It's very unfortunate and frustrating. I don't think I'm going to take any risks by using Codespaces ever again. Thank you for the response, nevertheless.

**Reply by cwndrws** [reply]
If putting a card on file is an option, it might be worth putting a card on file, setting your spending limit quite low, starting up your codespace, getting your stashed changes out of the codespace and then removing the payment method or setting your spending limit to $0 once you have your changes retrieved. It will cost a little bit of money, but not much and would probably be worth it depending on how much time it would take to recreate the changes.
